test/distributed/test_c10d_functional_native.py

```diff
@@ -565,14 +565,14 @@ def func(args: List[torch.Tensor]) -> torch.Tensor:
             # Expect no in-place with graph input (buf5, buf6 are clones)
             .check(
                 "torch.ops._c10d_functional.all_reduce_coalesced_"
-                ".default([buf5, buf6]"
+                ".default([buf15, buf16]"
             )
-            .check("torch.ops._c10d_functional.wait_tensor.default(buf0")
-            .check("torch.ops._c10d_functional.wait_tensor.default(buf1")
-            .check("torch.ops._c10d_functional.wait_tensor.default(buf5")
-            .check("torch.ops._c10d_functional.wait_tensor.default(buf6")
+            .check("torch.ops._c10d_functional.wait_tensor.default(buf15")
+            .check("torch.ops._c10d_functional.wait_tensor.default(buf16")
+            .check("torch.ops._c10d_functional.wait_tensor.default(buf15")
+            .check("torch.ops._c10d_functional.wait_tensor.default(buf16")
```

Comment:
This doesn't look right. Wait on buf0 and buf1 should take place.

Reply:
There are some additional manipulations with buffers as a result of additional getitem() ops, as with_effects() returns tuple.
But to my understanding it should not incur additional copies.
Will be good of course to get rid of those additional assignments.

Reply:
Oh I wasn't worried about the aesthetics. I was saying that the test doesn't verify buf0 and buf1 are waited, which are required to ensure correctness.